cmd/snap-confine/mount-support.c

```diff
@@ -412,12 +405,18 @@ static void sc_replicate_base_rootfs(const char *scratch_dir, const char *rootfs
             if (symlink(link_target, full_path) < 0) {
                 die("cannot create symbolic link \"%s\"", full_path);
             }
+            if (lchown(full_path, 0, 0) < 0) {
+                die("cannot change ownership for link \"%s\"", full_path);
+            }
         } else if (ent->d_type == DT_REG) {
             // Create an empty file which can be used as a mount point
             int fd = open(full_path, O_CREAT | O_TRUNC, 0644);
```

Comment:
The mode should not be 0644 here anymore.

Reply:
added a comment